ui: add Leader lease observability

nvb opened the pull request:
Fixes #125240.

This PR add observability for Leader leases to the DB console.

First, it adds information about Leader leases to the range details page.

<img width="796" alt="Screenshot 2024-07-17 at 5 06 34 PM" src="https://github.com/user-attachments/assets/860f8f67-dd58-4e48-b3ed-3cba1645cdae">
<img width="794" alt="Screenshot 2024-07-17 at 5 07 20 PM" src="https://github.com/user-attachments/assets/bae77afa-66e1-4a8b-b5a6-681260a194f2">

Next, it adds a "Lease Types" graph to Replication dashboard.

<img width="1004" alt="Screenshot 2024-07-18 at 11 45 00 AM" src="https://github.com/user-attachments/assets/dd5ae6ed-81fb-40bf-9dc1-11bf34971466">

Release note: None

nvb committed:
ui: add Leader lease observability

Fixes #125240.

This commit add observability for Leader leases to the DB console.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/127396)
<!-- Reviewable:end -->

nvb committed:
ui: add "Lease Types" graph to Replication dashboard

This commit adds a new "Lease Types" graph to the Replication dashboard.
The graph shows the number of leases of each type (expiration, epoch,
leader) held by the cluster over time.

Epic: None
Release note: None

nvanbenschoten commented:
@dhartunian I added one more small commit to the end of this PR. Mind taking a look when you get a chance?

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127396#-:-O2BWh_m56KvRQ4FPpwi:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten)



<!-- Sent from Reviewable.io -->

nvanbenschoten commented:
TFTRs!

bors r=dhartunian,arulajmani
